src/autolabel/tasks/classification.py

```diff
@@ -112,6 +113,53 @@ def get_explanation_prompt(self, example: Dict) -> str:
             labeled_example=fmt_example,
         )
 
+    def get_generate_dataset_prompt(
+        self, label: str, num_rows: int, guidelines: str = None
+    ) -> str:
+        pt = PromptTemplate(
+            input_variables=get_format_variables(self.GENERATE_DATASET_PROMPT),
+            template=self.GENERATE_DATASET_PROMPT,
+        )
+
+        # prepare task guideline
+        if guidelines is None:
+            guidelines = "You are an expert at generating plausible inputs for a given task.\n\nBEGIN TASK DESCRIPTION\n{task_guidelines}\nEND TASK DESCRIPTION"
+        labels_list = self.config.labels_list()
+        num_labels = len(labels_list)
+        fmt_task_guidelines = self.task_guidelines.format(
+            num_labels=num_labels, labels="\n".join(labels_list)
+        )
+        guidelines = guidelines.format(task_guidelines=fmt_task_guidelines)
+
+        # prepare columns
+        columns = get_format_variables(self.config.example_template())
+        columns.remove(self.config.label_column())
+
+        # prepare label descriptions
+        label_descriptions = f"Each input should fall into one of these {num_labels} categories. These are the only categories that the inputs can belong to."
```

Comment:
label descriptions are a good idea! lets also allow agent.run to use this and try to reuse code from that if possible